src/diagnostics/default_diagnostics.jl

```diff
@@ -0,0 +1,194 @@
+# This file is included by Diagnostics.jl and defines all the defaults for various models. A
+# model here is either a global AtmosModel, or small (sub)models (e.g., DryModel()).
+#
+# If you are developing new models, add your defaults here. If you want to add more high
+# level interfaces, add them here. Feel free to include extra files.
+
+"""
+    get_default_diagnostics(model)
+
+
+Return a list of `ScheduledDiagnostic`s associated with the given `model`.
+
+"""
+function get_default_diagnostics(model::AtmosModel)
+    # TODO: Probably not the most elegant way to do this...
+    defaults = Any[]
+
+    for field in fieldnames(AtmosModel)
+        def_model = get_default_diagnostics(getfield(model, field))
+        append!(defaults, def_model)
+    end
+
+    return defaults
+end
+
+# Base case: if we call get_default_diagnostics on something that we don't have information
+# about, we get nothing back (to be specific, we get an empty list, so that we can assume
+# that all the get_default_diagnostics return the same type). This is used by
+# get_default_diagnostics(model::AtmosModel), so that we can ignore defaults for submodels
+# that have no given defaults.
+get_default_diagnostics(_) = []
+
+
+"""
+    produce_common_diagnostic_function(period, reduction)
+
+
+Helper function to define functions like `get_daily_max`.
+"""
+function common_diagnostics(
+    period,
+    reduction,
+    output_writer,
+    long_names...;
+    pre_output_hook! = nothing,
+)
+    return [
+        ScheduledDiagnosticTime(
+            variable = ALL_DIAGNOSTICS[long_name],
+            compute_every = :timestep,
+            output_every = period, # seconds
+            reduction_time_func = reduction,
+            output_writer = output_writer,
+            pre_output_hook! = pre_output_hook!,
+        ) for long_name in long_names
+    ]
+end
+
+function average_pre_output_hook!(accum, counter)
+    @. accum = accum / counter
+    nothing
+end
+
+"""
+    daily_maxs(short_names...; output_writer = HDF5Writer())
+
+
+Return a list of `ScheduledDiagnostics` that compute the daily max for the given variables.
+"""
+daily_maxs(short_names...; output_writer = HDF5Writer()) =
+    common_diagnostics(24 * 60 * 60, max, output_writer, short_names...)
```

Comment:
Won't users generally want to have a single writer, so that we can efficiently read/write to disc? If a user calls any of these functions without the output writer, then what's the default behavior? I guess I'm wondering if we should just make `output_writer` non-optional.

Reply:
Yes, at the moment each diagnostic would get its own writer if you don't specify anything. But I think that I should work on this, and Simon also had comments on the writers.